.circleci/config.yml

```diff
@@ -381,31 +381,70 @@ jobs:
             equal: ["upload", "<< pipeline.parameters.SOURCE_TYPE >>"]
           steps:
             - run:
-                name: Download and Extract from S3
+                name: Download from S3 or IPFS based on URL
                 command: |
-                  aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
-                  aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
-                  aws configure set default.region $AWS_REGION
-
-                  if [ "<< pipeline.parameters.ENVIRONMENT >>" == "development" ]; then
-                    S3_BUCKET_NAME=$AWS_S3_USER_UPLOADS_BUCKET_NAME_DEV
-                  elif [ "<< pipeline.parameters.ENVIRONMENT >>" == "testnet-sepolia" ]; then
-                    S3_BUCKET_NAME=$AWS_S3_USER_UPLOADS_BUCKET_NAME_SEPOLIA
-                  elif [ "<< pipeline.parameters.ENVIRONMENT >>" == "testnet-sepolia-staging" ]; then
-                    S3_BUCKET_NAME=$AWS_S3_USER_UPLOADS_BUCKET_NAME_SEPOLIA_STAGING
+                  # Check if the URL is an IPFS URL
```

Comment:
@cheran-senthil would it be possible to do this based on download type eg zip vs folder?